src/modules/filesystem.ts

```diff
@@ -0,0 +1,109 @@
+import { dataDirectory, imagesDirectory } from '../config';
+import keyv from 'keyv';
+import { promises as fs, existsSync } from 'fs';
+import mimeTypes from 'mime-types';
+import path from 'path';
+import sharp from 'sharp';
+
+const cache = new keyv();
+
+export async function getTypes(): Promise<string[]> {
```

Comment:
This should be an enum.

Reply:
the types get evaluated dynamically by the contents of the /assets folder, not sure how we would solve that with an enum without having to change the source code to add new types